raiden/tests/unit/test_node_queue.py

```diff
@@ -43,9 +43,115 @@ def test_delivered_message_must_clean_unordered_messages(chain_id):
     )
 
     chain_state.queueids_to_queues[queue_identifier] = [first_message, second_message]
-    delivered_message = state_change.ReceiveDelivered(message_identifier)
+
+    delivered_message = state_change.ReceiveDelivered(recipient, message_identifier)
+
+    iteration = node.handle_delivered(chain_state, delivered_message)
+    new_queue = iteration.new_state.queueids_to_queues.get(queue_identifier, [])
+
+    assert first_message not in new_queue
+
+
+def test_delivered_processed_message_cleanup(chain_id):
+    pseudo_random_generator = random.Random()
+    block_number = 10
+    our_address = factories.make_address()
+    recipient = factories.make_address()
+    channel_identifier = 1
+    message_identifier = random.randint(0, 2 ** 16)
+    secret = factories.random_secret()
+
+    chain_state = state.ChainState(
+        pseudo_random_generator,
+        block_number,
+        our_address,
+        chain_id,
+    )
+    queue_identifier = QueueIdentifier(
+        recipient,
+        events.CHANNEL_IDENTIFIER_GLOBAL_QUEUE,
+    )
+
+    # Regression test:
+    # The code delivered_message handler worked only with a queue of one
+    # element
+    first_message = events.SendSecretReveal(
+        recipient,
+        channel_identifier,
+        message_identifier,
+        secret,
+    )
+    second_message = events.SendSecretReveal(
+        recipient,
+        channel_identifier,
+        random.randint(0, 2 ** 16),
+        secret,
+    )
+
+    # Register 2 messages into the queue
+    chain_state.queueids_to_queues[queue_identifier] = [first_message, second_message]
+
+    # Generate a random message_identifier
+    fake_message_identifier = random.randint(0, 2 ** 16)
+    node.inplace_delete_message(
+        chain_state,
```

Comment:
What I meant in my previous comment is this:

```python
recipient = factories.make_address()
channel_identifier = 1
secret = factories.random_secret()

msg1 = events.SendSecretReveal(
    recipient,
    channel_identifier,
    random.randint(0, 2 ** 16),
    secret,
)
msg2 = events.SendSecretReveal(
    recipient,
    channel_identifier,
    random.randint(0, 2 ** 16),
    secret,
)

message_queue = [msg1, msg2]
fake_message_identifier = random.randint(0, 2 ** 16)
node.inplace_delete_message(
    message_queue,
    state_change.ReceiveDelivered(recipient, fake_message_identifier),
)

assert msg1 in queueids_to_queues
assert msg2 in queueids_to_queues

message_queue = [msg1, msg2]
fake_recipient = factories.make_address()
node.inplace_delete_message(
    message_queue,
    state_change.ReceiveDelivered(fake_recipient, msg1.message_identifier),
)

assert msg1 in queueids_to_queues
assert msg2 in queueids_to_queues

message_queue = [msg1, msg2]
node.inplace_delete_message(
    message_queue,
    state_change.ReceiveDelivered(recipient, msg1.message_identifier),
)
assert msg1 not in queueids_to_queues
assert msg2 in queueids_to_queues
```

Just adding the function `inplace_delete_message` doesn't make the test easier to understand, but removing the extra objects (namely `ChainState`, `QueueIdentifier`), and using the a smaller unit to test `inplace_delete_message` instead of `node.handle_processed` does.